integrations/dump_restore_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2022 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package integrations
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"testing"
+
+	repo_model "code.gitea.io/gitea/models/repo"
+	"code.gitea.io/gitea/models/unittest"
+	user_model "code.gitea.io/gitea/models/user"
+	"code.gitea.io/gitea/modules/setting"
+	"code.gitea.io/gitea/modules/structs"
+	"code.gitea.io/gitea/modules/util"
+	"code.gitea.io/gitea/services/migrations"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDumpRestore(t *testing.T) {
+	onGiteaRun(t, func(t *testing.T, u *url.URL) {
+		AllowedDomains := setting.Migrations.AllowedDomains
+		setting.Migrations.AllowedDomains = "loopback"
+		AllowLocalNetworks := setting.Migrations.AllowLocalNetworks
+		setting.Migrations.AllowLocalNetworks = true
+		AppVer := setting.AppVer
+		setting.AppVer = "1.15"
```

Comment:
`AllowedDomains` I have a feeling that we do not need to set it. `AllowLocalNetworks` should be enough.

`AppVer` I see your comment, but I don't know where it happens, if there is a chance, I think we have better avoid the hard-coded version number. However if we really couldn't avoid it, I can accept.

Reply:
> AppVer I see your comment, but I don't know where the error happens,

It comes from the gitea SDK client the DumpRepository creates. The first thing it does is requesting the version from the API, tries to parse it using go-version and fails with the above error.

> if there is a chance, I think we have better avoid the hard-coded version number. However if we really couldn't avoid it, I can accept.

I agree. Is there a version number anywhere in the sources I could use instead?

> AllowedDomains I have a feeling that we do not need to set it. AllowLocalNetworks should be enough.

You are correct, removing them does not prevent tests from running.

---
[source](https://lab.fedeproxy.eu/forgefriends/forgefriends/-/merge_requests/30#note_683)

Reply:
Got it, `AppVer` seems another problem. The gitea-sdk need to parse it from server response. So let's ignore it in this PR.

Reply:
@dachary made changes to the patch to accommodate your suggestions regarding `AppVer`, please see the latest patch.